client/me/controller.js

```diff
@@ -192,15 +193,17 @@ export default {
 		analytics.pageView.record( basePath, ANALYTICS_PAGE_TITLE + ' > Notifications' );
 
 		React.render(
-			React.createElement( NotificationsComponent,
-				{
-					user: user,
-					userSettings: userSettings,
-					blogs: sites,
-					devices: devices,
-					path: context.path
-				}
-			),
+			React.createElement( Provider, { store: context.store }, () => {
```

Comment:
>If the root layout component is wrapped in `<Provider />`, why do we need to wrap `NotificationsComponent` in it here? Would upgrading to 0.14 obviate this?

It's unfortunately needed in both places, since there's technically two separate React render points. React 0.14 won't help here, though I think part of the idea with future plans for a global `<App />` component is to have a single render hierarchy.